service/worker/workerdeployment/workflow.go

```diff
@@ -765,25 +765,109 @@ func (d *WorkflowRunner) syncVersion(ctx workflow.Context, targetVersion string,
 }
 
 func (d *WorkflowRunner) syncUnversionedRamp(ctx workflow.Context, versionUpdateArgs *deploymentspb.SyncVersionStateUpdateArgs) error {
+	var err error
+	v := workflow.GetVersion(ctx, "syncUnversionedRamp", workflow.DefaultVersion, 1)
 	activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
-	var res deploymentspb.SyncUnversionedRampActivityResponse
-	err := workflow.ExecuteActivity(
-		activityCtx,
-		d.a.SyncUnversionedRamp,
-		&deploymentspb.SyncUnversionedRampActivityArgs{
-			CurrentVersion: d.State.RoutingConfig.CurrentVersion,
-			UpdateArgs:     versionUpdateArgs,
-		}).Get(ctx, &res)
-	if err != nil {
-		return err
+
+	if v == workflow.DefaultVersion {
+		var res deploymentspb.SyncUnversionedRampActivityResponse
+		err := workflow.ExecuteActivity(
+			activityCtx,
+			d.a.SyncUnversionedRamp,
+			&deploymentspb.SyncUnversionedRampActivityArgs{
+				CurrentVersion: d.State.RoutingConfig.CurrentVersion,
+				UpdateArgs:     versionUpdateArgs,
+			}).Get(ctx, &res)
+		if err != nil {
+			return err
+		}
+		// check propagation
+		err = workflow.ExecuteActivity(
+			activityCtx,
+			d.a.CheckUnversionedRampUserDataPropagation,
+			&deploymentspb.CheckWorkerDeploymentUserDataPropagationRequest{
+				TaskQueueMaxVersions: res.TaskQueueMaxVersions,
+			}).Get(ctx, nil)
+		if err != nil {
+			return err
+		}
+	} else {
+
+		// DescribeVersion activity to get all the task queues in the current version
+		var res deploymentspb.DescribeVersionFromWorkerDeploymentActivityResult
+		err := workflow.ExecuteActivity(
+			activityCtx,
+			d.a.DescribeVersionFromWorkerDeployment,
+			&deploymentspb.DescribeVersionFromWorkerDeploymentActivityArgs{
+				Version: d.State.RoutingConfig.CurrentVersion,
+			}).Get(ctx, &res)
+		if err != nil {
+			return err
+		}
+
+		// send in the task-queue families in batches of syncBatchSize
+		batches := make([][]*deploymentspb.SyncDeploymentVersionUserDataRequest_SyncUserData, 0)
+		syncReqs := make([]*deploymentspb.SyncDeploymentVersionUserDataRequest_SyncUserData, 0)
+
+		// Grouping by task-queue name
```

Comment:
Would you mind also deleting the line `// TODO (Carly): group by TQs by name and make only one sync req per name` from the activities.go file?